front/src/common/authorization/components/GrantsManagerSubjects.tsx

```diff
@@ -0,0 +1,140 @@
+import { useState } from 'react';
+
+import { ComboBox, Select } from '@osrd-project/ui-core';
+import cx from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import type { Grant, ResourceType, SubjectItemWithGrant } from 'common/api/mock/mockEditoastApi';
+import useSearchUser from 'common/useSearchUser';
+import { DEFAULT_GRANT } from 'modules/infra/consts';
+import generateGrantSelectProps from 'modules/infra/utils/generateGrantSelectProps';
+
+import useSubjectsResourceGrants from '../hooks/useSubjectsResourceGrants';
+
+type GrantsManagerSubjectsProps = {
+  resourceId: number;
+  resourceType: ResourceType;
+  userGrant: Grant;
+  resourceGrants?: {
+    [grant: string]: string[];
+  };
+};
+
+const GrantsManagerSubjects = ({
+  resourceId,
+  resourceType,
+  userGrant,
+  resourceGrants = {},
+}: GrantsManagerSubjectsProps) => {
+  const { t } = useTranslation('grantManagement');
+  const [displayUserSearchSection, setDisplayUserSearchSection] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<SubjectItemWithGrant[number] | undefined>();
+
+  const { userSubjectsList, updateUserResourceGrant } = useSubjectsResourceGrants({
+    resourceId,
+    resourceType,
+  });
+
+  const { searchResults, setSearchTerm, setSearchResults } = useSearchUser();
```

Comment:
It would be more explicit to rename `searchResults` into `searchedUsers` or `foundUsers`
```suggestion
  const { searchedUsers, setSearchTerm, setSearchResults } = useSearchUser();
```